File: examples/java/src/test/java/ai/marona/examples/ExamplesTest.java
package ai.marona.examples;

import ai.marona.Agent;
import ai.marona.Marona;
import ai.marona.ToolContext;
import com.fasterxml.jackson.databind.JsonNode;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.fasterxml.jackson.databind.node.ObjectNode;
import org.junit.jupiter.api.Test;

import java.util.ArrayList;
import java.util.List;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

class ExamplesTest {
    private static final ObjectMapper JSON = new ObjectMapper();

    @Test
    void basicAgentUsesProviderNeutralDefaults() {
        try (Marona marona = new Marona("test-key")) {
            Agent agent = BasicAgentExample.buildAgent(marona);
            assertEquals("Customer Assistant", agent.name());
            assertEquals("marona/auto", agent.model());
        }
    }

    @Test
    void multiAgentGraphBuildsWithoutNetworkAccess() {
        try (Marona marona = new Marona("test-key")) {
            Agent manager = MultiAgentExample.buildAgentGraph(marona);
            assertEquals("Customer Support Manager", manager.name());
        }
    }

    @Test
    void localFunctionToolUsesTrustedIdentityContext() {
        ObjectNode input = JSON.createObjectNode()
                .put("amount", 24)
                .put("duplicate_charge", true);
        ToolContext context = new ToolContext(
                "developer-1",
                "session-1",
                null,
                List.of("billing.refund.read"),
                List.of(),
                null
        );

        JsonNode result = MultiAgentExample.checkRefundTool().invoke(input, context);

        assertTrue(result.path("eligible").asBoolean());
        assertEquals(24, result.path("refund_amount").asInt());
        assertEquals("developer-1", result.path("customer_id").asText());
    }

    @Test
    void existingAgentAdapterPreservesIdentityContext() {
        List<String> calls = new ArrayList<>();
        BringYourOwnAgentExample.CapabilityConnection connection =
                new BringYourOwnAgentExample.CapabilityConnection() {
                    @Override
                    public List<JsonNode> listTools() {
                        return List.of(JSON.createObjectNode().put("name", "example__lookup"));
                    }

                    @Override
                    public JsonNode callTool(
                            String name,
                            JsonNode input,
                            String userId,
                            String conversationId
                    ) {
                        calls.add(name + ":" + userId + ":" + conversationId);
                        return JSON.createObjectNode().put("ok", true);
                    }
                };
        BringYourOwnAgentExample.ExistingAgentAdapter adapter =
                new BringYourOwnAgentExample.ExistingAgentAdapter(connection);

        JsonNode result = adapter.executeSelectedTool(
                "example__lookup",
                JSON.createObjectNode().put("query", "Marona"),
                "developer-1",
                "session-1"
        );

        assertTrue(result.path("ok").asBoolean());
        assertEquals(List.of("example__lookup:developer-1:session-1"), calls);
    }
}

File: examples/typescript/test/examples.test.ts
import assert from "node:assert/strict";
import test from "node:test";

import type { MCPTool } from "marona";

import { buildAgent } from "../src/basic-agent.js";
import { ExistingAgentAdapter, type CapabilityConnection } from "../src/bring-your-own-agent.js";
import { buildAgentGraph, checkRefund } from "../src/multi-agent.js";
import { buildRealtimeAgent } from "../src/realtime-agent.js";

process.env.MARONA_API_KEY = "test-key";
delete process.env.MARONA_APP_SLUG;

test("basic Agent uses provider-neutral defaults", async () => {
  const { agent } = await buildAgent();
  assert.equal(agent.name, "Customer Assistant");
  assert.equal(agent.model, "marona/auto");
  assert.equal(agent.toolChoice, "auto");
});

test("multi-Agent graph contains delegation and handoff", () => {
  const { manager } = buildAgentGraph();
  assert.equal(manager.name, "Customer Support Manager");
  assert.equal(Array.isArray(manager.tools) && manager.tools.length, 1);
  assert.equal(manager.handoffs.length, 1);
  assert.equal(checkRefund.name, "check_refund");
});

test("existing Agent adapter preserves identity context", async () => {
  const calls: unknown[] = [];
  const tools: MCPTool[] = [
    {
      name: "example__lookup",
      description: "Lookup",
      inputSchema: { type: "object" },
      annotations: {},
    },
  ];
  const connection: CapabilityConnection = {
    listTools: () => tools,
    callTool: async (name, input, options) => {
      calls.push({ name, input, options });
      return { ok: true };
    },
  };
  const adapter = new ExistingAgentAdapter(connection);
  assert.equal(adapter.availableTools()[0]?.name, "example__lookup");
  assert.deepEqual(
    await adapter.executeSelectedTool(
      "example__lookup",
      { query: "Marona" },
      { userId: "developer-1", sessionId: "session-1" },
    ),
    { ok: true },
  );
  assert.deepEqual(calls, [
    {
      name: "example__lookup",
      input: { query: "Marona" },
      options: { userId: "developer-1", conversationId: "session-1" },
    },
  ]);
});

test("realtime Agent declares audio contract", () => {
  const { agent } = buildRealtimeAgent();
  assert.equal(agent.voice?.turn_detection, "semantic_vad");
  assert.deepEqual(agent.modalities, ["text", "audio"]);
  assert.deepEqual(agent.outputModalities, ["text", "audio"]);
});
